### ISSUE TITLE
Option to fail on warnings

### ISSUE THREAD

**Issue by l1bbcsg** [google/closure-compiler#3451]
I would like to propose an option that could make the compiler fail if at least one warning is encountered, regardless of errors and exit with a non-zero exit code.

This could be a simple flags such as `--fail-on-warnings` or a more sophisticated setting like `--error-level` that could introduce more granular control, i.e. still exit with 0 if compilation failed with errors.

Use case for this is either a purist approach when no warnings are tolerated or code quality control CI job that verifies new code with `--checks-only`.

**Comment by brad4d** [comment]
I believe this option already exists. The magic words are `--jscomp_error=*` to say "treat all diagnostic groups as errors".

**Comment by l1bbcsg** [comment]
I don't think this is quite the right option. It seems to change behaviour much more elevating a lot of other warnings that are not normally reported to errors.

This example:

```js
/**
 * @type {number}
 */
let a = 10;

a = 'foo'
```

Produces a warning:

```sh
google-closure-compiler a.js --checks_only --warning_level verbose
```

```
a.js:6: WARNING - [JSC_TYPE_MISMATCH] assignment
found   : string
required: number
a = 'foo'
^^^^^^^^^

0 error(s), 1 warning(s), 100.0% typed
```

But with `--jscomp_error=*` there are now two errors:

```sh
google-closure-compiler a.js --checks_only --warning_level verbose --jscomp_error=*
```

```
a.js:6: ERROR - [JSC_MISSING_SEMICOLON] Missing semicolon
a = 'foo'
^^^^^^^^^

a.js:6: ERROR - [JSC_TYPE_MISMATCH] assignment
found   : string
required: number
a = 'foo'
^^^^^^^^^

2 error(s), 0 warning(s), 100.0% typed
```

On a large code base I encountered many more new errors and compiler actually crashed with `java.lang.RuntimeException: INTERNAL COMPILER ERROR`.

**Comment by brad4d** [comment]
Created internal issue
http://b/139437890

**Comment by brad4d** [comment]
This is looking harder to do than I'd wish, but I think the gist is that the proposed option would need to trigger doing `options.addWarningsGuard(new StrictWarningsGuard())` when the compiler options are created in `CommandLineRunner#createOptions()`.

**Comment by kaovilai** [comment]
Bump. Really wanted to use it with GitHub Actions for CI. Currently all exits regardless of error/warning levels are zero exits. If anyone have a bash workaround that detects a non-zero length output and fail
